src/query/expression/src/schema.rs

```diff
@@ -165,6 +176,25 @@ pub struct DataField {
     computed_expr: Option<ComputedExpr>,
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum VariantType {
```

Comment:
better to rename this as `VariantDataType`, because we already have a struct named `VariantType`, same name can cause confusion.